jobclient/python/__init__.py

```diff
@@ -0,0 +1,206 @@
+               application: Optional[Application] = None) -> UUID:
+        """Submit a single job to Cook.
+
+        Required Parameters
+        -------------------
+        :param command: The command to run on Cook.
+        :type command: str
+        :param cpus: The number of CPUs to request from Cook.
+        :type cpus: float
+        :param mem: The amount of memory, in GB, to request from Cook.
+        :type mem: float
+        :param max_retries: The number of times this job should be retried
+            before failing.
+        :type max_retries: int
+        Optional Parameters
+        -------------------
+        :param uuid: The UUID of the job to submit. If this value is not
+            provided, then a random UUID will be generated.
+        :type uuid: UUID, optional
+        :param env: Environment variables to set within the job's context,
+            defaults to None.
+        :type env: Dict[str, str], optional
+        :param labels: Labels to assign to the job, defaults to None.
+        :type labels: Dict[str, str], optional
+        :param max_runtime: The maximum number seconds this job should be
+            allowed to run, defaults to None.
+        :type max_runtime: int, optional
+        :param name: A name to assign to the job, defaults to None.
+        :type name: str, optional
+        :param priority: A priority to assign to the job, defaults to None.
+        :type priority: int, optional
+        :param application: Application information to assign to the job,
+            defaults to None.
+        :type application: Application, optional
+        Output
+        ------
+        :return: The UUID of the newly-created job.
+        :rtype: UUID
+        """
+        uuid = str(uuid if uuid is not None else util.make_temporal_uuid())
+        payload = {
+            'command': command,
+            'cpus': cpus,
+            'mem': mem,
+            'uuid': uuid,
+            'max-retries': max_retries
+        }
+        if env is not None:
+            payload['env'] = env
+        if labels is not None:
+            payload['labels'] = labels
+        if max_runtime is not None:
+            payload['max-runtime'] = max_runtime
+        if name is not None:
+            payload['name'] = name
+        if priority is not None:
+            payload['priority'] = priority
+        if application is not None:
+            payload['application'] = application.to_dict()
+        payload = {'jobs': [payload]}
+        url = urlunparse(('http', self.__netloc, self.__job_endpoint, '', '', ''))  # noqa E501
+        _LOG.debug(f"Sending POST to {url}")
+        _LOG.debug("Payload:")
+        _LOG.debug(json.dumps(payload, indent=4))
+        resp = requests.post(url, json=payload)
+        if not resp.ok:
+            _LOG.error(f"Could not submit job: {resp.status_code} {resp.text}")
+            resp.raise_for_status()
+
+        return UUID(uuid)
+
+    def query(self, uuid: UUID) -> Job:
```

Comment:
I think it would be nice to also allow just a string for the UUID.

Reply:
:+1: